src/eth_provider/provider.rs

```diff
@@ -655,7 +655,14 @@ where
         // unwrap option
         let to: kakarot_core::core::Option = {
             match request.to {
-                Some(to) => kakarot_core::core::Option { is_some: FieldElement::ONE, value: into_via_wrapper!(to) },
+                Some(to) => match to {
+                    TxKind::Create => {
+                        kakarot_core::core::Option { is_some: FieldElement::ZERO, value: FieldElement::ZERO }
+                    }
+                    TxKind::Call(to) => {
+                        kakarot_core::core::Option { is_some: FieldElement::ONE, value: into_via_wrapper!(to) }
+                    }
+                },
                 None => kakarot_core::core::Option { is_some: FieldElement::ZERO, value: FieldElement::ZERO },
             }
```

Comment:
we can probably collapse this into something like: 
```rust
            match request.to {
                Some(TxKind::Call(to)) => kakarot_core::core::Option { is_some: FieldElement::ONE, value: into_via_wrapper!(to) },
                _ => kakarot_core::core::Option { is_some: FieldElement::ZERO, value: FieldElement::ZERO },
            }
```

Reply:
Yes nice, just fixed